### PR TITLE
Cli 169 log streams

### PR DESCRIPTION
**Pull request by bright-poku**
### Description

This PR adds the following logs streams command

- `auth0 logs streams create`
- `auth0 logs streams update`
- `auth0 logs streams list`
- `auth0 logs streams show`
- `auth0 logs streams delete`
- `auth0 logs streams open`

Screen Shot 2021-05-28 at 9 21 10 AM

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [X ] The correct base branch is being used, if not `master`

### CODE REVIEW

File: internal/auth/auth.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ var requiredScopes = []string{
 	"read:connections", "update:connections",
 	"read:client_keys", "read:logs", "read:tenant_settings", "read:custom_domains",
 	"read:anomaly_blocks", "delete:anomaly_blocks",
+	"create:log_streams", "delete:log_streams", "read:log_streams", "update:log_streams",
```
Comment (cyx) [inline internal/auth/auth.go:39]: There's a small test for these in `auth_test.go`. Do you mind adding `log_streams` to the list of CRUD resources? (https://github.com/auth0/auth0-cli/blob/main/internal/auth/auth_test.go#L7-L12)

File: internal/display/log_streams.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	r.logStreamResult(logs)
+}
+
+func (r *Renderer) logStreamResult(logs *management.LogStream) {
+	r.Result(&logStreamView{
+		ID:     ansi.Faint(logs.GetID()),
+		Name:   logs.GetName(),
+		Type:   logs.GetType(),
+		Status: logs.GetStatus(),
+		//Sink: logs.Sink,
```
Comment (cyx) [inline internal/display/log_streams.go:85]: Can we remove it if we don't need it?

File: internal/display/log_streams.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	}
+}
+
+func (v *logStreamView) KeyValues() [][]string {
+	return [][]string{
+		[]string{"ID", ansi.Faint(v.ID)},
+		[]string{"NAME", v.Name},
+		[]string{"TYPE", v.Type},
+		[]string{"STATUS", v.Status},
+		//[]string{"SINK", fmt.Sprintln(v.Sink)},
```
Comment (cyx) [inline internal/display/log_streams.go:36]: Ditto on removing if not needed.

File: internal/cli/log_streams.go [hunk 4]
```diff
@@ -0,0 +1,823 @@
...
+	}
+	logStreamType = Flag{
+		Name:       "Type",
+		LongForm:   "type",
+		ShortForm:  "t",
+		Help:       "Type of the log stream. Possible values: http, eventbridge, eventgrid, datadog, splunk, sumo.",
+		IsRequired: true,
+	}
+	typeOptions = []string{
+		"Http",
```
Comment (cyx) [inline internal/cli/log_streams.go:44]:
```suggestion
		"HTTP",
```

File: internal/cli/log_streams.go [hunk 5]
```diff
@@ -0,0 +1,823 @@
...
+		"Eventbridge",
+		"Eventgrid",
+		"Datadog",
```
Comment (cyx) [inline internal/cli/log_streams.go:47]:
```suggestion
		"EventBridge",
		"EventGrid",
		"DataDog",
```

File: internal/cli/log_streams.go [hunk 6]
```diff
@@ -0,0 +1,823 @@
...
+		"Splunk",
+		"Sumo",
+	}
+	httpEndpoint = Flag{
+		Name:       "Http Endpoint",
```
Comment (cyx) [inline internal/cli/log_streams.go:52]:
```suggestion
		Name:       "HTTP Endpoint",
```

File: internal/cli/log_streams.go [hunk 7]
```diff
@@ -0,0 +1,823 @@
...
+		LongForm:   "http-endpoint",
+		Help:       "HTTP endpoint.",
+		IsRequired: true,
+	}
+	httpContentType = Flag{
+		Name:         "Http Content Type",
```
Comment (cyx) [inline internal/cli/log_streams.go:58]:
```suggestion
		Name:         "HTTP Content Type",
```

File: internal/cli/log_streams.go [hunk 8]
```diff
@@ -0,0 +1,823 @@
...
+		LongForm:     "http-type",
+		Help:         "HTTP Content-Type header. Possible values: application/json.",
+		AlwaysPrompt: true,
+	}
+	httpContentFormat = Flag{
+		Name:         "Http Content Format",
```
Comment (cyx) [inline internal/cli/log_streams.go:64]:
```suggestion
		Name:         "HTTP Content Format",
```

File: internal/cli/log_streams.go [hunk 9]
```diff
@@ -0,0 +1,823 @@
...
+		LongForm:     "http-format",
+		Help:         "HTTP Content-Format header. Possible values: JSONLINES, JSONARRAY, JSONOBJECT.",
+		AlwaysPrompt: true,
+	}
+	httpAuthorization = Flag{
+		Name:         "Http Authorization",
```
Comment (cyx) [inline internal/cli/log_streams.go:70]:
```suggestion
		Name:         "HTTP Authorization",
```

**Review by cyx — APPROVED** [review]
Just a bunch of non-blocking nits, but otherwise :shipit: LGTM

File: internal/cli/log_streams.go [hunk 10]
```diff
@@ -0,0 +1,823 @@
...
+	var inputs struct {
+		ID   string
+		Type string
+	}
+
+	cmd := &cobra.Command{
+		Use:   "show",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Show a log stream by id",
+		Long:  "Show an application.",
```
Comment (Widcket) [inline internal/cli/log_streams.go:205]:
```suggestion
		Short: "Show a log stream by Id",
		Long:  "Show a log stream by Id.",
```

File: internal/cli/log_streams.go [hunk 11]
```diff
@@ -0,0 +1,823 @@
...
+			}
+
+			a := &management.LogStream{ID: &inputs.ID}
+
+			if err := ansi.Waiting(func() error {
+				var err error
+				a, err = cli.api.LogStream.Read(inputs.ID)
+				return err
+			}); err != nil {
+				return fmt.Errorf("Unable to load application. The Id %v specified doesn't exist", inputs.ID)
```
Comment (Widcket) [inline internal/cli/log_streams.go:225]:
```suggestion
				return fmt.Errorf("Unable to load log stream. The Id %v specified doesn't exist", inputs.ID)
```

File: internal/cli/log_streams.go [hunk 12]
```diff
@@ -0,0 +1,823 @@
...
+				inputs.ID = args[0]
+			}
+
+			// Load log stream by id
+			if err := ansi.Waiting(func() error {
+				var err error
+				current, err = cli.api.LogStream.Read(inputs.ID)
+				return err
+			}); err != nil {
+				return fmt.Errorf("Unable to load logstreams. The Id %v specified doesn't exist", inputs.ID)
```
Comment (Widcket) [inline internal/cli/log_streams.go:490]:
```suggestion
				return fmt.Errorf("Unable to load logstream. The Id %v specified doesn't exist", inputs.ID)
```

**Review by Widcket — APPROVED** [review]
Awesome. Thanks for going above and beyond!

**Pull request merged**